### PR TITLE
Rz dedup

### PR DESCRIPTION
**Pull request by rzinke**

### CODE REVIEW

**Comment by rzinke** [comment]
This PR deconflicts, and replaces https://github.com/aria-tools/ARIA-tools/pull/454.

**Comment by rzinke** [comment]
At the present time, ARIA products are only available with 30 m (0.000277778 arcsec) or 90 m (0.000833334) resolution based on [ISCE2 Dockerized topsApp processing](https://github.com/ACCESS-Cloud-Based-InSAR/DockerizedTopsApp/blob/5bf4e32a2b0f7e6dffc90c90e79da01441f1d5e3/isce2_topsapp/topsapp_proc.py#L56). Rounding errors can cause unintended deviations from those pixel sizes. https://github.com/aria-tools/ARIA-tools/pull/458/commits/9ce610a75c8a6879fcc8ba392d639e9e9dc84a1d and https://github.com/aria-tools/ARIA-tools/pull/458/commits/dcdd435d4227d6312a2b5d8baa428a5864753c23 determine the nearest intended pixel size based on the detected pixel size (e.g., 0.000833329 -> 0.000833334 arcsec).

File: tools/ARIAtools/extractProduct.py [hunk 1]
```diff
@@ -564,8 +566,9 @@ def merged_productbbox(
         ds = None
 
     # Adjust arrres to supported resolution
-    res_ndx = np.argmin([np.abs(p-px_size) for px_size in ARIA_PX_SIZES])
-    arrres = ARIA_PX_SIZES[res_ndx]
+    for i, res in enumerate(arrres):
+        res_ndx = np.argmin([np.abs(res-px_size) for px_size in ARIA_PX_SIZES])
+        arrres[i] = ARIA_PX_SIZES[res_ndx]
```
Comment (sssangha) [inline tools/ARIAtools/extractProduct.py:595]: Elegant, clean workaround for this!

**Comment by sssangha** [comment]
@rzinke , I've been stress testing your PR and a lot of it runs smoothly!

when I rerun `ariaTSSetup.py` and try to extract a correction layer (e.g. SET), I have a couple of observations:
1.  The DEM is redownloaded if not explicitly specified in the command line, even if it's already there from the previous run. This behavior is only circumvented if you explicitly specify the path to the existing DEM. It looks like similar logic applies to the mask file.


2. The cutline slightly changes, and is reflected as so in the `config.yaml` file each time a correction file is iteratively extracted:

Before `-l 'solidEarthTide'`:
```
prods_TOTbbox: POLYGON ((-118.523377673576 37.0627282073644, -117.556660049369 37.1987893479971,
  -117.49828629251 36.925169162822, -117.498194552768 36.9251813880188, -117.498168419725
  36.9250588779455, -116.647377383485 37.0384202172966, -116.413033592732 35.8781694965187,
  -116.41296811948 35.8781784086941, -116.412965790614 35.8781668768844, -116.412767624527
  35.8781938476329, -116.399316842636 35.8111094046942, -116.375859901649 35.6941202339577,
  -116.341784825989 35.5254124095775, -116.342026462833 35.525379307661, -116.17999000071
  34.7172384958184, -116.179952255829 34.7172437419007, -116.179950411858 34.7172345441569,
  -116.179822583571 34.7172523084529, -116.153884705814 34.5870409221744, -116.109188396571
  34.3641225051396, -116.109373162402 34.364096666849, -116.109325710082 34.3638584207445,
  -116.109424396982 34.3638446182278, -115.878213541876 33.2031372446911, -116.302813243568
  33.1425321117082, -116.767626235979 33.0761954866964, -116.790260144933 33.184013085308,
  -117.721296355817 33.0444911999459, -117.782334110021 33.3198693264883, -118.668114504917
  33.1808244590334, -118.978724406925 34.5131341113245, -119.253611382046 35.6754548960399,
  -119.57595518474 37.0177594693425, -118.547982963561 37.170951516264, -118.523496232554
  37.0628220926193, -118.523401937738 37.0628353658255, -118.523377673576 37.0627282073644))
```

vs after `-l 'solidEarthTide'`:
```
prods_TOTbbox: POLYGON ((-118.523377673576 37.0627282073644, -117.556660049369 37.1987893479971,
  -117.49828629251 36.925169162822, -117.498194552768 36.9251813880188, -117.498168419725
  36.9250588779455, -116.647377383485 37.0384202172966, -116.413033592732 35.8781694965187,
  -116.41296811948 35.8781784086941, -116.412965790614 35.8781668768844, -116.412767624527
  35.8781938476329, -116.399316842636 35.8111094046942, -116.375859901649 35.6941202339577,
  -116.341784825989 35.5254124095775, -116.342026462833 35.525379307661, -116.17999000071
  34.7172384958184, -116.179952255829 34.7172437419007, -116.179950411858 34.7172345441569,
  -116.179822583571 34.7172523084529, -116.153884705814 34.5870409221744, -116.109188396571
  34.3641225051396, -116.109373162402 34.364096666849, -116.109325710082 34.3638584207445,
  -116.109424396982 34.3638446182278, -115.878213541876 33.2031372446911, -116.302813243568
  33.1425321117082, -116.767626235979 33.0761954866964, -116.790260144933 33.184013085308,
  -117.721296355817 33.0444911999459, -117.782334110021 33.3198693264883, -118.668114504917
  33.1808244590334, -118.978724406925 34.5131341113245, -119.253611382046 35.6754548960399,
  -119.57595518474 37.0177594693425, -118.547982963561 37.170951516264, -118.523496232554
  37.0628220926193, -118.523401937738 37.0628353658255, -118.523377673576 37.0627282073644))
```
  
  
  vs after `-l 'troposphereTotal' `:
  ```
  prods_TOTbbox: POLYGON ((-118.523377673576 37.0627282073644, -117.556660049369 37.1987893479971,
  -117.49828629251 36.925169162822, -117.498194552768 36.9251813880188, -117.498168419725
  36.9250588779455, -116.647377383485 37.0384202172966, -116.413033592732 35.8781694965187,
  -116.41296811948 35.8781784086941, -116.412965790614 35.8781668768844, -116.412767624527
  35.8781938476329, -116.410291641074 35.8658451253035, -116.375859901684 35.6941202341304,
  -116.341784825989 35.5254124095775, -116.342026462833 35.5253793076609, -116.17999000071
  34.7172384958184, -116.179952255829 34.7172437419007, -116.179950411858 34.7172345441569,
  -116.179822583571 34.7172523084529, -116.153884705814 34.5870409221744, -116.109188396571
  34.3641225051396, -116.109373162402 34.364096666849, -116.109325710082 34.3638584207445,
  -116.109424396982 34.3638446182278, -115.878213541876 33.2031372446911, -116.302813243568
  33.1425321117082, -116.767626235979 33.0761954866964, -116.790260144933 33.184013085308,
  -117.721296355817 33.0444911999459, -117.782334110021 33.3198693264883, -118.668114504917
  33.1808244590334, -118.978724406938 34.5131341113771, -119.253611382017 35.6754548959206,
  -119.57595518474 37.0177594693425, -118.547982963561 37.170951516264, -118.523496232554
  37.0628220926193, -118.523401937738 37.0628353658255, -118.523377673576 37.0627282073644))
  ```
  
  This is also reflected in the `coherence` VRT layers, which are overwritten. Everything else is untouched.
  
  
To replicate my test, you can run the following:
```
# Download products
ariaDownload.py -t 64 -b '34.5 35.7 -116.5 -114.2' -s 20180101 -e 20180201 -o download

# Extract products
ariaTSsetup.py -f "products/*.nc" -w test_dir --verbose --log-level debug


# Extract SET
ariaTSsetup.py -f "products/*.nc" -w test_dir --verbose --log-level debug -l 'solidEarthTide'

# Extract tropo
# We can circumvent the DEM redownload by explicitly specifying the path as below
ariaTSsetup.py -f "products/*.nc" -w test_dir --verbose --log-level debug -l 'troposphereTotal' -tm HRRR -d test_dir/DEM/glo_90.dem
```

**Review by sssangha — APPROVED** [review]
LGTM!

**Pull request merged**